src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -375,6 +375,9 @@ public static void resolveUserAndExecute(
      * @throws Exception exception
      */
     private static boolean checkUserPermissions(User requestedUser, User resourceUser, String workflowId) throws Exception {
+        if (requestedUser == null || resourceUser == null) {
+            return false;
+        }
         if (resourceUser.getBackendRoles() == null || requestedUser.getBackendRoles() == null) {
             return false;
         }
```

Comment:
These lines can probably be combined into one conditional, or maybe we can create a helper method taking varargs `User...` that we can iterate over and just call `if (nullOrNoRoles(requestedUser, resourceUsers)) { return false; }`.